### PR TITLE
[runtime] Add option to always use posix locale on darwin

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by alexanderkyte** [comment]
Supports https://github.com/mono/mono/issues/9925

**Comment by kumpera** [comment]
Under what scenarios should the posix locale differ from the Darwin one?

File: mono/metadata/locales.c [hunk 1]
```diff
@@ -559,7 +559,8 @@ get_current_locale_name (void)
 #ifdef HOST_WIN32
 	locale = g_win32_getlocale ();
 #elif defined (__APPLE__)	
-	locale = get_darwin_locale ();
+	if (!g_hasenv ("MONO_DARWIN_FAST_LOCALE"))
```
Comment (akoeplinger) [inline mono/metadata/locales.c]: I'd name this `MONO_DARWIN_PREFER_POSIX_LOCALE`.

We'll also need to document it in the man page.

Comment (migueldeicaza) [reply mono/metadata/locales.c]: The POSIX locale is based on environment variables and might not be set when launching outside of a shell (launchd, the open command, iOS apps).

**Comment by migueldeicaza** [comment]
@kumpera The POSIX locale is based on environment variables and might not be set when launching outside of a shell (launchd, the open command, iOS apps).

**Pull request closed without merging**